components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java

```diff
@@ -187,12 +198,18 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) {
                 }
 
                 if (serviceProvider != null) {
```

Comment:
This check is done at line 179 already. I think we can remove this

Reply:
ack